src/main.rs

```diff
@@ -17,18 +17,45 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+use clap::{CommandFactory, ErrorKind, Parser};
+
 use astarte_device_sdk::options::AstarteOptions;
 use astarte_device_sdk::registration;
 use astarte_device_sdk::AstarteDeviceSdk;
 
 use astarte_message_hub::config::MessageHubOptions;
 use astarte_message_hub::error::AstarteMessageHubError;
 
+/// A central service that runs on (Linux) devices for collecting and delivering messages from N
+/// apps using 1 MQTT connection to Astarte.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// Path to a valid .toml file containing the message hub configuration.
+    /// Do not provide this option with the `store_directory` option.
+    #[clap(short, long, value_parser)]
+    toml: Option<String>,
+    /// Directory used by the HTTP and ProtoBuf servers to retain the provided configuration.
+    /// Do not use this option with the `toml`  option.
+    #[clap(short, long, value_parser)]
+    store_directory: Option<String>,
+}
+
 #[tokio::main]
 async fn main() -> Result<(), AstarteMessageHubError> {
     env_logger::init();
+    let args = Args::parse();
+
+    if args.toml.is_some() && args.store_directory.is_some() {
```

Comment:
This can be achieved with [Arg::conflict_with](https://docs.rs/clap/3.2.23/clap/builder/struct.Arg.html#method.conflicts_with).

You could pass it to both toml and store_directory as a macro attribute.